collect_app/src/test/java/org/odk/collect/android/formentry/audit/FormSaveViewModelTest.java

```diff
@@ -353,6 +353,24 @@ public void whenFormSaverFinishes_mediaManagerIsCleared() {
         assertThat(recentFiles.isEmpty(), equalTo(true));
     }
 
+    @Test
+    public void whenFormSaverFinishes_onRecreatingFormSaveViewModel_mediaManagerIsCleared() {
+        viewModel.markOriginalFileOrDelete("index", "blah");
+        viewModel.replaceRecentFileForQuestion("index", "blah");
+
+        FormSaveViewModel restoredViewModel = new FormSaveViewModel(savedStateHandle, () -> CURRENT_TIME, formSaver, mediaUtils, null);
+        restoredViewModel.formLoaded(formController);
+
+        restoredViewModel.saveForm(Uri.parse("file://form"), true, "", false);
+        whenFormSaverFinishes(SaveFormToDisk.SAVED);
+
+        Map<String, String> originalFiles = savedStateHandle.get(FormSaveViewModel.ORIGINAL_FILES);
+        Map<String, String> recentFiles = savedStateHandle.get(FormSaveViewModel.RECENT_FILES);
+
+        assertThat(originalFiles.isEmpty(), equalTo(true));
```

Comment:
I think this should check that the files are deleted etc (so check the correct calls are made to `mediaUtils`) rather than checking the `savedStateHandle`).

Reply:
Actually so maybe we only need `markOriginalFileOrDelete_whenQuestionIndexHasAnswer_onRecreatingViewModel_deletesFile`? Unless there are reasons to check `savedStateHandle` directly.